packages/core-data/src/actions.js

```diff
@@ -287,7 +280,7 @@ export function* editEntityRecord( kind, name, recordId, edits, options = {} ) {
 		}, {} ),
 		transientEdits,
 	};
-	return {
+	return await dispatch( {
```

Comment:
In my book, the `editEntityRecord` action is completely synchronous. It synchronously loads the entity from store, doesn't wait for resolution (quite the opposite, it throws when the entity is not already loaded), and then synchronously applies edits to the in-memory store.